Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,199 @@
+    }
+
+    AZStd::string SdfAssetBuilder::CreateDefaultProcPrefab(
+        const AssetBuilderSDK::ProcessJobRequest& request,
+        [[maybe_unused]] AssetBuilderSDK::ProcessJobResponse& response) const
+    {
+        // The following code is similar to DefaultProceduralPrefabGroup::CreatePrefabGroupManifestUpdates().
+        // It might be possible to refactor the two to have shared logic.
+        // It's not worth investigating until the SdfAssetBuilder is fully functional though, because the code
+        // is likely to change pretty drastically still.
+
+        // Create a prefab name based on the source file name.
+        AZStd::string prefabTemplateName{ request.m_sourceFile };
+        AzFramework::StringFunc::Path::ReplaceExtension(prefabTemplateName, "procprefab");
+        AZ::StringFunc::Replace(prefabTemplateName, "\\", "/"); // the source folder uses forward slash
+
+        // clear out any previously created prefab template for this path
+        auto* prefabSystemComponentInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
+        AzToolsFramework::Prefab::TemplateId prefabTemplateId =
+            prefabSystemComponentInterface->GetTemplateIdFromFilePath({ prefabTemplateName.c_str() });
+        if (prefabTemplateId != AzToolsFramework::Prefab::InvalidTemplateId)
+        {
+            prefabSystemComponentInterface->RemoveTemplate(prefabTemplateId);
+            prefabTemplateId = AzToolsFramework::Prefab::InvalidTemplateId;
+        }
+
+        // create a default entity in the prefab to verify functionality.
+        AZ::EntityId entityId;
+        AzToolsFramework::EntityUtilityBus::BroadcastResult(
+            entityId, &AzToolsFramework::EntityUtilityBus::Events::CreateEditorReadyEntity, "Test");
+
+        AZStd::vector<AZ::EntityId> entities = { entityId };
+
+        // create prefab from the "set" of entities (currently just the single default entity)
+        AzToolsFramework::Prefab::PrefabSystemScriptingBus::BroadcastResult(
+            prefabTemplateId,
+            &AzToolsFramework::Prefab::PrefabSystemScriptingBus::Events::CreatePrefabTemplate,
+            entities,
+            prefabTemplateName);
+
+        if (prefabTemplateId == AzToolsFramework::Prefab::InvalidTemplateId)
+        {
+            AZ_Error("prefab", false, "Could not create a prefab template for entities.");
+            return {};
+        }
+
+        // Convert the prefab to a JSON string
+        AZ::Outcome<AZStd::string, void> outcome;
+        AzToolsFramework::Prefab::PrefabLoaderScriptingBus::BroadcastResult(
+            outcome,
+            &AzToolsFramework::Prefab::PrefabLoaderScriptingBus::Events::SaveTemplateToString,
+            prefabTemplateId);
+
+        if (outcome.IsSuccess() == false)
+        {
+            AZ_Error("prefab", false, "Could not create JSON string for template; maybe NaN values in the template?");
```

Comment:
Nan values?

I believe this comment should be updated to account for SDFormat builder
```suggestion
            AZ_Error("SDFormat Builder", false, "Could not serialize prefab template as a JSON string");
```

Reply:
Lol, yeah. I already changed the error tag to SdfAssetBuilderName, but I wasn't exactly sure why the original error message from `DefaultProceduralPrefabGroup::CreatePrefabGroupManifestUpdates()` was warning about NaNs so I left it there. I'll change the message to your proposed message though, it makes more sense. :)